protocol/opcm-v2.md

```diff
@@ -0,0 +1,234 @@
+- **Backwards compatibility**: Both functions support all v1 operations
+- **Release readiness**: Unified initialization eliminates upgrade function maintenance
+
+## Alternatives Considered
+
+### Alternative 1: Continue Incremental Improvements to OPCM v1
+
+**Approach**: Continue the pattern of making incremental improvements to OPCM v1—adding features, refactoring specific pain points, and addressing issues as they arise without a larger architectural change.
+
+**Advantages**:
+- Lower risk, familiar codebase
+- No migration complexity
+
+**Disadvantages**:
+- We've already done significant incremental work and are hitting diminishing returns
+- Can't address core architectural issues (function proliferation, hardcoded assumptions) without larger changes
+- Would continue accumulating technical debt
+
+**Rejection rationale**: We've reached the limit of what incremental improvements can achieve. The current pain points—especially around dynamic game types and unified initialization—require larger architectural changes. Continuing with incremental fixes would mean accepting the existing limitations indefinitely. We know the right direction now, so it's time to make the larger change.
+
+### Alternative 2: Complete OPCM Redesign (Different Architecture)
+
+**Approach**: Rather than improving OPCM's existing design, start from scratch with a fundamentally different architecture.
+
+**Advantages**:
+- Could potentially optimize for currently unknown future requirements
+
+**Disadvantages**:
+- OPCM's general design is sound—the issues are specific pain points, not fundamental flaws
+- Would throw away working, well-understood code
+- Much higher risk and longer timeline
+- Harder to maintain backwards compatibility
+
+**Rejection rationale**: The core OPCM design is good. We don't want to replace it; we want to improve specific aspects. A complete redesign would be throwing out the baby with the bathwater. The v2 approach preserves what works while fixing what doesn't.
+
+## Risks and Uncertainties
+
+### Security Risk: Increased Flexibility Creates Increased Attack Surface
+
+**Risk**: While OPCM v2 isn't a massive architectural change, it is still a change to critical infrastructure. The increased flexibility (particularly around dynamic game types and re-initialization) could introduce new security vulnerabilities if not implemented defensively.
+
+**Mitigation**: 
+- Comprehensive security audit with focus on the new flexible interfaces
+- Defensive coding practices throughout implementation
+- State diff review for the first OPCM v2 upgrade to verify correctness
+- Multiple rounds of internal security review before audit
+
+### Functionality Coverage Risk: Missing OPCM v1 Capabilities
+
+**Risk**: We might inadvertently miss some existing functionality that OPCM v1 provides when transitioning to the new 2-function interface.
+
+**Mitigation**:
+- Ensure all existing OPCM v1 tests pass with v2 implementation
```

Comment:
Are we planning to use the '[strangler pattern](https://learn.microsoft.com/en-us/azure/architecture/patterns/strangler-fig#solution)' here, so that both OPCMs exist and are tested in parallel during development of V2? This was used successfully when migrating the deploy scripts to the new pattern without the I/O contracts.